pandas/core/indexes/period.py

```diff
@@ -594,13 +595,13 @@ def get_indexer_non_unique(self, target):
         return ensure_platform_int(indexer), missing
 
     def _get_unique_index(self, dropna=False):
-        """
-        wrap Index._get_unique_index to handle NaT
-        """
-        res = super()._get_unique_index(dropna=dropna)
-        if dropna:
-            res = res.dropna()
-        return res
+        if self.is_unique and not dropna:
+            return self
+
+        result = self._data.unique()
```

Comment:
why is this not self.unique()?

Reply:
Because we want to pass an EA to shallow_copy

Reply:
question here

Reply:
calling self.unique returns an Index, while we want a PeriodArray to pass to _shallow_copy below